react/components/ExtraData.tsx

```diff
@@ -185,7 +198,7 @@ const ExtraData: React.VFC<Props> = ({
       </span>
 
       <form onSubmit={handleSubmit}>
-        {cardType === 'new' && (
+        {mustShowDocumentField && (
```

Comment:
we can't _not show_ the document in case it isn't required, I suggest creating a new prop on the `Payment` component to hide this document field if it isn't required (something like `hideOptionalDocument`)

Reply:
But isn't the behavior of checkout v6?